app/lib/actions/katello/repository/create_container_push_root.rb

```diff
@@ -0,0 +1,44 @@
+module Actions
+  module Katello
+    module Repository
+      class CreateContainerPushRoot < Actions::EntryAction
+        def plan(root, relative_path = nil)
+          repository = ::Katello::Repository.new(:environment => root.organization.library, :content_view_version => root.organization.library.default_content_view_version, :root => root)
+          #Container push may concurrently call root add several times before the db can update.
+          # If the root already exists, we can skip the creation of the root and repository.
+          # We acquire a lock on the product to ensure that the root is not created multiple times by different workers.
+
+          root.product.with_lock do
+            begin
+              root.save!
+            rescue ActiveRecord::RecordInvalid => e
+              if root.is_container_push && e.message.include?("Name has already been taken for this product")
+                Rails.logger.warn("Skipping root repository creation as container push root repository already exists: #{root.container_push_name}")
```

Comment:
What do you think about moving this to debug logging? `warn` implies something went wrong, but this is a normal bit of functionality. As a warning it might contribute to filling up logs.

Reply:
And same for the log below.